File: android/core/domain/src/main/java/com/enterprise/core/domain/repositories/FeedRepository.kt
package com.enterprise.core.domain.repositories

import com.enterprise.core.domain.entities.DomainResult
import com.enterprise.core.domain.entities.UIComponent
import kotlinx.coroutines.flow.Flow

/** Dynamic feed repository contract (SSOT: UI observes local DB). */
interface FeedRepository {
    fun getDynamicFeed(): Flow<DomainResult<List<UIComponent>>>
    suspend fun refreshFeed(): DomainResult<List<UIComponent>>
    suspend fun getCachedFeed(): List<UIComponent>
}

File: ios/EnterpriseMobileApp/Sources/Core/Domain/Repositories/FeedRepository.swift
import Foundation

/// Repository protocol for the dynamic feed (Server-Driven UI).
public protocol FeedRepository: Sendable {
    func getDynamicFeed() -> AsyncStream<DomainResult<[UIComponent]>>
    func refreshFeed() async -> DomainResult<[UIComponent]>
    func getCachedFeed() async -> [UIComponent]
}
